### ISSUE TITLE
Migrate to 2017.2 MP protocol

### ISSUE THREAD

**Issue by onox** [c172p-team/c172p-detailed#1011]
In the 2017.2 MP protocol we have about access to about 90 booleans, 40 generic floats, and 80 short ints.

We should make a list of all the properties that are transmitted as integers, but are actually used like boolean (on/off). We could use a JSBSim `<switch>` instead of FG's autopilot/property-rule XML to control the booleans.

See http://wiki.flightgear.org/Multiplayer_protocol#Properties_Packet

See https://github.com/onox/p51d/blob/master/p51d-jsbsim-set.xml#L171-L232 and https://github.com/onox/p51d/blob/master/Systems/effects.xml#L166-L175 for an example.

**Comment by wlbragg** [comment]
So we can simply change any of the int's to bool where appropriate?

>We could use a JSBSim <switch> instead of FG's autopilot/property-rule XML to set a boolean.

I'm not quite following this one. Aren't we using switch in this manner already? Can you direct me to a situation where we are not?

**Comment by onox** [comment]
I meant some of the `<filter>`'s in `Systems/ground-effects.xml`.

**Comment by onox** [comment]
> So we can simply change any of the int's to bool where appropriate?

Yes, but many of the XML files for the effects have to be changed. In those files we have a `<condition>` with a `<greater-than>` (for example) in it. That `<greater-than>` can then be removed.

**Comment by onox** [comment]
If we use a JSBSim `<output>` property (in `Systems/c172p-ground-effects`), then we can use/alias that property in `<multiplayer><generic>` and in the flight recorder. And we don't need those filters in `Systems/ground-effects.xml` anymore then.

**Comment by onox** [comment]
Hey @dany93, I noticed on the [FG forums](https://forum.flightgear.org/viewtopic.php?f=4&t=28279&p=329936) you mentioned the P-51D. I've worked on it for a while in my [repostory](https://github.com/onox/p51d).

**Comment by dany93** [comment]
Yes, I had seen that (I don't remember how and when).

As the aircraft had serious issues in Nov. 2015 (did not open, engine did not start), I wished to ask and propose a few changes to Hal (among them, one Autostart from efel). I wondered why he did not react. That's when I learned (and announced in the US forum) that Hal V. Engel had passed away.

Since then, I've make it a bit milder at takeoff and landing, changed the spiraling propwash formulation, moved the markers on the dashboard (they were outside their right place). Maybe a few other small things.
More importantly I discovered the reason of an old, not understood, erratic bad bug, which made it crash in unclear circumstances at engine start. It was a bug in the core FG code, fixed by Bertrand Coconnier after my observations.
These changes have been included in SVN with the help of Ludovic Brenta.

I don't know if your changes can be included in SVN (with not too many conflicts) by the same way or your own rights, you are much better than me on this.

**Comment by onox** [comment]
They already are. Stuart merged the changes back to fgaddon.

**Comment by onox** [comment]
@wlbragg I've started working on refactoring the MP properties. If there are any effects missing in the list above, feel free to add it.

I've moved several properties from `/sim/model/c172p/securing` to `/aircraft/securing` because navigating to `/sim/model/c172p/` in the property dialog is just a big pain.
